### ISSUE TITLE
Switching from pysqlchipher3 to sqlcipher3

### ISSUE THREAD

**Issue by dylanljones** [dylanljones/pyrekordbox#87]
Hey everyone,

I am currently working on the migration to SqlAlchemy 2.0. It seems like the currently used [pysqlchipher3](https://github.com/rigglemania/pysqlcipher3) package is using an too old version of SQLite. I found a more up-to-date implementation: [sqlcipher3](https://github.com/coleifer/sqlcipher3). This slightly changes the installation steps on Windows, which I am updating now. Can anyone confirm if this package also works on MacOS? Do the install instructions (except the GitHub URL) have to be changed?

```shell
git clone https://github.com/coleifer/sqlcipher3
cd sqlcipher3
SQLCIPHER_PATH=$(brew info sqlcipher | awk 'NR==4 {print $1; exit}'); C_INCLUDE_PATH="$SQLCIPHER_PATH"/include LIBRARY_PATH="$SQLCIPHER_PATH"/lib python setup.py build
SQLCIPHER_PATH=$(brew info sqlcipher | awk 'NR==4 {print $1; exit}'); C_INCLUDE_PATH="$SQLCIPHER_PATH"/include LIBRARY_PATH="$SQLCIPHER_PATH"/lib python setup.py install
```

Would really appreciate some help on updating the installation docs!

**Comment by dylanljones** [comment]
Hmm, didn't run into the issue with the space in the path of the OpenSSL library, just checked my `PATH` and `OPENSSL_CONF` variables and it points to Program Files. But i think it would help to clarify that in the docs. Also thanks for the hint with the 8.1 formatting, didn't know that one:D 
And yes, we should probably switch to the newest version of VS, i didn't update that in years to be honest since i never use it...
Migrating to `sqlcipher3` only fixes the issues with the quotes, sadly...
Thanks for your help!

**Comment by dylanljones** [comment]
ah never mind, got it to work:)

**Comment by dylanljones** [comment]
I compared the two packages and think there shouldn't be any problem. Closing this issue for now

**Comment by dadrake3** [comment]
Ive tried the osx instructions using sqlcipher3 and It doesn't seem to work, I keep getting the sqlalchemy.exc.DatabaseError: (sqlite3.DatabaseError) file is not a database error which indicates sqlalchemy is not trying to decrypt the db properly I believe

**Comment by dadrake3** [comment]
I have an m1 Mac and I believe this is part of this issue with building sqlcipher3

python setup.py build seems to be failing silently and I cant properly import it once running install so Im guessing this is causing sqlalchemy to fallback to the regular sqlite engine

ill keep digging

**Comment by dadrake3** [comment]
@dylanljones yes i was able to get pysqlcipher3 to install. I think part of the problem is im using an m1 mac.

Using the arm64 version of sqlcipher from brew and an arm64 conda env, i get missing header file errors when building and installing both pysqlcipher3 and sqlcipher (I had to use —force on the build step otherwise it fails silently)

when i switched to the x86 version of sqlcipher from x86 brew and used an x86 conda env I was able to successfully build and install pyscipher3. When I tried for sqlcipher3 the build appears to be successful (i.e no errors like before) but it doesn’t actually link the c binary and I can’t import it.

I opened a ticket on sqlcipher3 but the maintainer was less than helpful lol.

also side note when I pip install rekordbox it wants to default to sqlalchemy 1.4 rather than 2.0 but idt this is related since the behavior described above occurred both for pip installs and when I built from requirements.txt.

**Comment by dylanljones** [comment]
Thats weird, but i too had some problems building `sqlcipher3` against the amalgamation on Windows. Sorry you are having issues, but the switch is kind of mandatory since SqlAlchemy 2.0 doesn't work with `pysqlcipher3` and it doesn't seem like that package is still maintained. Hope we can figure this out soon for M1 macs. I just saw the issue you created, kind of had to laugh at the response of the maintainer:D He could be a little more helpful...

SqlAlchemy 2.0 will be used from version 0.2.1 on-wards, so the changes aren't on pip yet. If you install from GitHub it should install SqlAlchemy2.0 though!

**Comment by dadrake3** [comment]
ahh makes sense, Ill see if I can get sqlcipher3 to build correctly and then give you an update, hopefully I can do it without the need for the x86 libraries too because switching brews and condas isn't ideal lol

**Comment by dadrake3** [comment]
@dylanljones Hey forgot to follow up, my bad. I was able to build and install sqlcipher3 against my home-brew install by adding some symlinks. Pyrekordbox also works properly now too

these 2 did the trick

```
ln -s /opt/homebrew/lib/libsqlcipher.a /usr/local/lib/libsqlcipher.a
ln -s /opt/homebrew/include/sqlcipher /usr/local/include/sqlcipher
```
